src/NuGet.Core/NuGet.LibraryModel/LibraryDependency.cs

```diff
@@ -99,7 +99,8 @@ public LibraryDependency Clone()
                 },
                 SuppressParent = SuppressParent,
                 Type = Type,
-                AutoReferenced = AutoReferenced
+                AutoReferenced = AutoReferenced,
+                NoWarn = NoWarn
```

Comment:
also, shouldn't you be cloning this list itself? both objects will have a reference to the same list now